src/components/modals/MigrateAccounts/steps/03-step-currency.js

```diff
@@ -0,0 +1,193 @@
+// @flow
+
+import invariant from 'invariant'
+import React, { PureComponent, Fragment } from 'react'
+import logger from 'logger'
+import { Trans } from 'react-i18next'
+
+import type { Account } from '@ledgerhq/live-common/lib/types/account'
+import { addAccounts } from '@ledgerhq/live-common/lib/account'
+import { DeviceShouldStayInApp } from '@ledgerhq/errors'
+import Text from 'components/base/Text'
+import TrackPage from 'analytics/TrackPage'
+
+import Button from 'components/base/Button'
+import Box from 'components/base/Box'
+import { CurrencyCircleIcon } from 'components/base/CurrencyBadge'
+import { getCurrencyBridge } from 'bridge'
+import type { StepProps } from '../index'
+
+type Props = StepProps & { replaceAccounts: (Account[]) => void }
+
+// $FlowFixMe
+const remapTransportError = (err: mixed, appName: string): Error => {
+  if (!err || typeof err !== 'object') return err
+
+  const { name, statusCode } = err
+
+  const errorToThrow =
+    name === 'BtcUnmatchedApp' || statusCode === 0x6982 || statusCode === 0x6700
+      ? new DeviceShouldStayInApp(null, { appName })
+      : err
+
+  return errorToThrow
+}
+
+class StepCurrency extends PureComponent<Props> {
+  componentDidMount() {
+    this.props.setScanStatus('scanning')
+    this.startScanAccountsDevice()
+  }
+
+  componentWillUnmount() {
+    this.unsub()
+  }
+
+  scanSubscription = null
+
+  unsub = () => {
+    if (this.scanSubscription) {
+      this.scanSubscription.unsubscribe()
+    }
+  }
+
+  startScanAccountsDevice() {
+    this.unsub()
+    const {
+      currency,
+      device,
+      setScanStatus,
+      addMigratedAccount,
+      replaceAccounts,
+      transitionTo,
+    } = this.props
+    try {
+      invariant(currency, 'No currency to scan')
+      invariant(device, 'No device')
+
+      const bridge = getCurrencyBridge(currency)
+
+      // TODO: use the real device
+      const devicePath = device.path
+      this.scanSubscription = bridge.scanAccountsOnDevice(currency, devicePath).subscribe({
+        next: account => {
+          const { migratableAccounts } = this.props
+          if (
+            migratableAccounts[currency.id] &&
+            migratableAccounts[currency.id].some(
+              a => a.freshAddressPath === account.freshAddressPath,
```

Comment:
mmh I would not trust that to be stable and source of trust for migration. Is there a way we can either use a method in live-common that delegate this work? I think it was findAccountMigration job somehow.

NB in future, we might have migration that doesn't have guarantee of fresh address equality, imagine on bitcoin as this is not immutable.

we need to me to see why we need `addMigratedAccount `

there is a dangerous assumption that this will cover the exact diff that the `addAccounts` logic is doing and I would suggest we do some kind of "diff" using it.